lib/upload.js

```diff
@@ -531,6 +534,40 @@ class BaseUpload {
       return
     }
 
+    if (this.options.enableChecksum) {
+      const optionsReq = this._openRequest('OPTIONS', this.options.endpoint)
```

Comment:
Currently, we do not check extension support for any extension that is used by tus-js-client. We assume that the user knows that the server supports the extension if they enable it. I would be in favor of keeping in line with this style and against making an OPTIONS request here.